### ISSUE TITLE
Speed test with partial fail

### ISSUE THREAD

**Issue by tiansh** [soerface/7billionhumans#19]
I had made a solution for Lv38 which is both shorter and faster than existed solution. But it not produce correct result every time. Should I add it? Maybe we should also keep the _correct_ one?

Size: 14, Speed: 29, Pass: 18/25

```
a:
step n
if c <= mem1 or
 mem1 != datacube:
	mem1 = set c
endif
if n == wall:
	pickup mem1
	mem4 = nearest shredder
	b:
	mem2 = nearest worker
	if mem2 == nothing:
		giveto mem4
	endif
	if myitem >= mem2 or
	 myitem == nothing:
		mem4 = nearest hole
		step mem4
	endif
	jump b
endif
jump a

```

**Comment by ansvonwa** [comment]
First, I'd suggest to keep all working solutions (discussed with @soerface yesterday, he agreed).

Second, the problem you mention is even a bit worse: There are astonishing many solutions that do not work in _some_ edge cases. One could consider them as 'buggy' and discard them, but it is likely that we never find those edge cases, so we cannot really distinguish between those that seem to work always and those really do.
And - as you know - the game accepts solutions with >50% success rate, so it is intentional that we may produce 'buggy' solutions.
Thus, I would suggest to also check-in solutions that only work sometimes.

This leads to the next problem: We should add comments about the success rate if we know it is not exactly 100% - and about the number of tests that we did. (I have solutions that only work with low chances (about 50%), but are far smaller/faster than the "global best".)

TL;DR
My suggestion: add it, keep the old one, add a comment "-- Success Rate: 18/25"

**Comment by soerface** [comment]
I merged @ansvonwa suggestion of keeping a "Success Rate: x/y" comment (https://github.com/soerface/7billionhumans/blob/master/add_solution.py#L36). We need to think about how to display such solutions on the website, but I think it is fine as long as the success rate is above 50%.
